cas-config/.gitignore

```diff
@@ -0,0 +1 @@
+callback.py
```

Comment:
Can you explain this?

Reply:
Sorry for not answering earlier, we had a *long* string of holidays. I'll get back to you with the rest of it in a few hours, but I think I'll explain this one right now:

The idea is that should the user need to introduce custom code that fires on every CAS login, they would be able to do this without forking at all by dropping in a file -- presumably, here, and naming it `callback.py`. This is driven by experience: Many of our changes were entirely unsuitable for inclusion into upstream, *(chiefly because our use case was not suitable for the platform, but that's a problem not of my making)* but it was impossible to implement them without forking, which results in a maintenance nightmare.

I.e. this is where the CAS attributes delivered in an ad-hoc arbitrary format would turn into course subscriptions. I guess I'll document that...